File: sdks/ts/src/sdk/models/shared/collaborations.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { Collaboration } from "./collaboration";


export enum CollaborationsOrderDirectionEnum {
    Asc = "ASC",
    Desc = "DESC"
}


// CollaborationsOrder
/** 
 * The order in which a pagination is ordered
**/
export class CollaborationsOrder extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=by" })
  by?: string;

  @SpeakeasyMetadata({ data: "json, name=direction" })
  direction?: CollaborationsOrderDirectionEnum;
}


// Collaborations
/** 
 * The part of an API response that describes pagination
**/
export class Collaborations extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=entries", elemType: Collaboration })
  entries?: Collaboration[];

  @SpeakeasyMetadata({ data: "json, name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "json, name=offset" })
  offset?: number;

  @SpeakeasyMetadata({ data: "json, name=order", elemType: CollaborationsOrder })
  order?: CollaborationsOrder[];

  @SpeakeasyMetadata({ data: "json, name=total_count" })
  totalCount?: number;
}

File: java-sdk/lib/src/main/java/models/shared/Collaborations.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * Collaborations
 * The part of an API response that describes pagination
**/
public class Collaborations {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("entries")
    public Collaboration[] entries;
    public Collaborations withEntries(Collaboration[] entries) {
        this.entries = entries;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("limit")
    public Long limit;
    public Collaborations withLimit(Long limit) {
        this.limit = limit;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("offset")
    public Long offset;
    public Collaborations withOffset(Long offset) {
        this.offset = offset;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("order")
    public CollaborationsOrder[] order;
    public Collaborations withOrder(CollaborationsOrder[] order) {
        this.order = order;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("total_count")
    public Long totalCount;
    public Collaborations withTotalCount(Long totalCount) {
        this.totalCount = totalCount;
        return this;
    }
}